tox.ini

```diff
@@ -0,0 +1,117 @@
+[tox]
+envlist =
+    py{36,37,38}-test{,-alldeps,-oldestdeps,-devdeps,-numpy116,-numpy117,-numpy118}{,-cov}
+    build_docs
+    linkcheck
+    codestyle
+requires =
+    setuptools >= 30.3.0
+    pip >= 19.3.1
+isolated_build = true
+
+[testenv]
+
+# Pass through the following environemnt variables which are needed for the CI
+passenv = HOME WINDIR LC_ALL LC_CTYPE CC CI TRAVIS TEST_READ_HUGE_FILE
+
+# For coverage, we need to pass extra options to the C compiler
+setenv =
+    cov: CFLAGS = --coverage -fno-inline-functions -O0
+
+# Run the tests in a temporary directory to make sure that we don't import
+# astropy from the source tree
+changedir = .tmp/{envname}
+
+# tox environments are constructued with so-called 'factors' (or terms)
+# separated by hyphens, e.g. test-devdeps-cov. Lines below starting with factor:
+# will only take effect if that factor is included in the environment name. To
+# see a list of example environments that can be run, along with a description,
+# run:
+#
+#     tox -l -v
+#
+description =
+    run tests
+    alldeps: with all optional dependencies
+    devdeps: with the latest developer version of key dependencies
+    oldestdeps: with the oldest supported version of key dependencies
+    cov: and test coverage
+    numpy116: with numpy 1.16.*
+    numpy117: with numpy 1.17.*
+    numpy118: with numpy 1.18.*
+
+deps =
+
+    numpy116: numpy==1.16.*
+    numpy117: numpy==1.17.*
+    numpy118: numpy==1.18.*
+
+    # The oldestdeps factor is intended to be used to install the oldest versions of all
+    # dependencies that have a minimum version
+    oldestdeps: numpy==1.16.*
+    oldestdeps: matplotlib==2.1.*
+    oldestdeps: asdf==2.4.*
+
+    # The devdeps factor is intended to be used to install the latest developer version
+    # of key dependencies.
+    devdeps: git+https://github.com/numpy/numpy.git#egg=numpy
+    devdeps: git+https://github.com/matplotlib/matplotlib.git#egg=matplotlib
+    devdeps: git+https://github.com/spacetelescope/asdf.git#egg=asdf
+
+# The following indicates which extras_require from setup.cfg will be installed
+extras =
+    test
+    alldeps: all
+
+commands =
+    pip freeze
+    !cov: pytest --pyargs astropy {toxinidir}/docs {posargs}
+    cov: pytest --pyargs astropy {toxinidir}/docs --cov astropy --cov-config={toxinidir}/.coveragerc {posargs}
```

Comment:
Question: If we include `.coveragerc` file in `package_data`, does that mean we don't need to specify `--cov-config` anymore?

Reply:
We still need to specify ``--cov-config`` in all cases because pytest isn't run at the root of the repository.